lightkurve/correctors/metrics.py

```diff
@@ -0,0 +1,333 @@
+"""Metrics to assess under- and over-fitting of systematic noise.
+
+This module provides two metrics, `overfit_metric_lombscargle` and `underfit_metric_neighbors`,
+which enable users to assess whether the noise in a systematics-corrected light curve has been
+under- or over-fitted.  These features were contributed by Jeff Smith (cf. https://github.com/KeplerGO/lightkurve/pull/855)
+and are in turn inspired by similar metrics in use by the PDC module of the official Kepler/TESS pipeline.
+"""
+import logging
+
+import numpy as np
+from scipy.interpolate import PchipInterpolator
+
+from .. import LightCurve
+
+log = logging.getLogger(__name__)
+
+
+def overfit_metric_lombscargle(
+    original_lc: LightCurve, corrected_lc: LightCurve, n_samples: int = 10
+) -> float:
+    """Uses a LombScarglePeriodogram to assess the change in broad-band
+    power in a corrected light curve to measure the degree of over-fitting.
+
+    The to_periodogram Lomb-Scargle method is used and the sampling band is
+    from one frequency separation to the Nyquist frequency
+
+    This over-fitting goodness metric is calibrated such that a metric
+    value of 0.5 means the introduced noise due to over-fitting is at the
+    same power level as the uncertainties in the light curve.
+
+    Parameters
+    ----------
+    original_lc : LightCurve
+        Uncorrected light curve.
+    corrected_lc : LightCurve
+        Light curve from which systematics have been removed.
+    n_samples : int
+        The number of times to compute and average the metric
+        This can stabilize the value, default = 10
+
+    Returns
+    -------
+    overfit_metric : float
+        A float in the range [0,1] where 0 => Bad, 1 => Good
+    """
+    # The fit can sometimes result in NaNs
+    # Also ignore masked cadences
+    # Also median normalize original and correctod LCs
+    orig_lc = original_lc.copy()
+    orig_lc = orig_lc.remove_nans().normalize()
```

Comment:
Correct.  I want to prevent the concept of the `Corrector.cadence_mask` property from leaking outside of the `Corrector` class. (where `cadence_mask` is used to flag cadences that ought to be ignored while fitting the noise model).

When calling the standalone metric functions, a user can apply a mask to the input light curves as necessary, i.e. one can do `overfit_metric(original_lc[mask], corrected_lc[mask])`, so no need for a separate mask keyword there.

Reply:
(But we do indeed need to remove the outdated comment on line 47!)

Reply:
(I went ahead and removed the outdated comment on line 47 which read `# Also ignore masked cadence`.)